app.py

```diff
@@ -1,13 +1,91 @@
-from flask import Flask
-import requests
+from flask import Flask, render_template, redirect, request
+from flask_login import LoginManager, login_user, logout_user, current_user, login_required
+from flask_mongoengine import MongoEngine
+from flask_mongoengine.wtf import model_form
+from wtforms import PasswordField
+
 
 app = Flask(__name__)
+app.config["DEBUG"] = True
+app.config['MONGODB_SETTINGS'] = { 'db' : 'users' }
+app.config['SECRET_KEY'] = 'JJAM'
+app.config['WTF_CSRF_ENABLED'] = True
+
+db = MongoEngine(app)
+login_manager = LoginManager()
+login_manager.init_app(app)
+
+
+
+
+class User(db.Document):
+  name = db.StringField(required = True, unique = True)
+  password = db.StringField(required = True)
+  courses = db.StringField()
+  def is_authenticated(self):
+    users = User.objects(name = self.name, password = self.password, courses = self.courses)
+    return len(users) != 0
```

Comment:
I'm not sure I follow the logic here.

If we have a `User` class, wouldn't it automatically be stored in the database (and have `len(users) == 1`)?